content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+
+- **Building as root:** building packages require you to run melange as root, thus the `--privileged` parameter.
+- **Volumes:** here we set up **two** volumes: the current folder is shared with the /work location in the container, so that the packages generated will be available in your local system at the ./packages location. The second volume shares the `/tmp/melange_out` location in the container with your `/tmp` folder, so that you can have access to melange's working files even when the build fails. This helps a lot with debugging!
+- **Architecture:** limit the build to your own system's architecture while you are developing the package, this will make the build faster and avoid issues.
+- **Signing key:** remember this key will be required later as well, when you install the package in a container image using apko.
+
+Here is the full command:
+
+```shell
+docker run --privileged --rm -v "${PWD}":/work -v /tmp/melange_out:/tmp \
+  cgr.dev/chainguard/melange build my-package.yaml \
+  --arch x86_64 \
+  --signing-key melange.rsa
+```
+
+### When the build fails
+It is likely that your build won't work on the first run, and that is completely normal because there are many moving parts and hidden dependencies when building packages from source. Check the errors and iterate until you have a successful build. Use `set -x` before commands in your pipeline to get extended debug information:
+
+```yaml
+  - name: Install Composer
+    runs: |
+      set -x
+      EXEC_DIR="${{targets.destdir}}/usr/bin"
+      mkdir -p "${EXEC_DIR}"
+      mv composer.phar "${EXEC_DIR}/composer"
+      chmod +x "${EXEC_DIR}/composer"
+```
+
+Most build issues are caused by missed dependencies, even when the error message might be misleading. Another common reason for build errors are wrong file or directory paths. The [melange documentation](https://edu.chainguard.dev/open-source/melange/troubleshooting/) has more pointers to help with debugging, in case you need it.
+
+As mentioned before, there might be cases where you'll need to first build a dependency, and then use this dependency to build the package you need.
+
+When working with local dependencies, use the following notation in your `packages` list, inside the `environment` section:
+
+```yaml
+environment:
+  contents:
+    repositories:
+      - https://packages.wolfi.dev/bootstrap/stage3
+      - https://packages.wolfi.dev/os
+      - '@local /work/packages'
+    keyring:
+      - https://packages.wolfi.dev/bootstrap/stage3/wolfi-signing.rsa.pub
+      - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+    packages:
+      - busybox
+      - mypackage@local
+```
+
+This will look for a package named "mypackage" in your local packages/ folder.
+
+### When the build is successful
+First of all, celebrate!: 🎉 Check the packages folder, you should see a directory for each built architecture (in my case I get x86_64) with your built apks (package + subpackages) AND an `APKINDEX.tar.gz` file:
```

Comment:
```suggestion
First of all, celebrate! 🎉

Check the packages folder, you should find a directory for each built architecture (in my case I get x86_64) with your built apks (package + subpackages) along with an `APKINDEX.tar.gz` file:
```

So I can handle a *little* informality :smile: I wanted to even highlight it a bit more here by adding a line break

Reply:
hahaha I was worried about that part 👀 😸